src/java/org/apache/cassandra/repair/autorepair/AutoRepairUtils.java

```diff
@@ -600,18 +783,27 @@ else if (!hostIdsInCurrentRing.contains(nodeHistory.hostId))
                                 "First node in priority list is {}", ClusterMetadata.current().directory.addresses.get(NodeId.fromUUID(priorityHostId)));
                     return NOT_MY_TURN;
                 }
+
```

Comment:
Note that if a node is marked for priority, it will ignore this check, which I think seems reasonable to me!